docs/aws/audit/redshiftmonitoring/rules/emr_cluster_master_public_ip_remediation.mdx

```diff
@@ -1,6 +1,269 @@
+
+   ```json
+   {
+     "Classification": "emrfs-site",
+     "Properties": {
+       "fs.s3.consistent.retryPeriodSeconds": "10",
+       "fs.s3.consistent": "true",
+       "fs.s3.consistent.retryCount": "5"
+     }
+   }
+   ```
+
+   Then, create the security configuration using the AWS CLI:
+
+   ```sh
+   aws emr create-security-configuration --name MySecurityConfig --security-configuration file://emr-security-config.json
+   ```
+
+2. **Create a Subnet Without Public IPs:**
+   Ensure you have a subnet that does not assign public IPs by default. You can create a subnet in a VPC and disable the auto-assign public IP setting.
+
+   ```sh
+   aws ec2 create-subnet --vpc-id vpc-12345678 --cidr-block 10.0.1.0/24 --availability-zone us-west-2a
+   ```
+
+   Disable auto-assign public IP:
+
+   ```sh
+   aws ec2 modify-subnet-attribute --subnet-id subnet-12345678 --no-map-public-ip-on-launch
+   ```
+
+3. **Launch the EMR Cluster in the Private Subnet:**
+   When creating the EMR cluster, specify the subnet that does not assign public IPs. Use the `--ec2-attributes` parameter to specify the subnet ID.
+
+   ```sh
+   aws emr create-cluster --name "MyEMRCluster" --release-label emr-5.30.0 --applications Name=Hadoop Name=Spark --ec2-attributes SubnetId=subnet-12345678 --instance-type m5.xlarge --instance-count 3 --use-default-roles
+   ```
+
+4. **Verify the Configuration:**
+   After creating the cluster, verify that the master node does not have a public IP. You can describe the cluster and check the master node's network interfaces.
+
+   ```sh
+   aws emr describe-cluster --cluster-id j-1K48XXXXXXHCB
+   ```
+
+   Check the `MasterPublicDnsName` field in the output. It should be empty if the master node does not have a public IP.
+
+By following these steps, you can ensure that the EMR Cluster Master Node does not have a public IP, thereby enhancing the security of your EMR cluster.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the EMR Cluster Master Node from having a public IP in AWS using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Below are the steps to achieve this:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+
+3. **Configure EMR Cluster**:
+   When creating an EMR cluster, ensure that the `EmrManagedMasterSecurityGroup` is configured to not allow public IPs.
+
+4. **Create EMR Cluster with No Public IP for Master Node**:
+   Use the Boto3 `emr` client to create an EMR cluster with the appropriate configuration.
+
+Here is a Python script to achieve this:
+
+```python
+import boto3
+
+# Initialize a session using your AWS credentials
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+    aws_secret_access_key='YOUR_AWS_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>